### PR TITLE
7 configuration provide advanced settings

### PR DESCRIPTION
**Pull request by catorghans**
#3, #5, #6, #7 combined. Please have a look.

### CODE REVIEW

File: src/Hook/NodeViewAlterPostRenderHook.php [hunk 1]
```diff
@@ -19,11 +19,19 @@ public static function trustedCallbacks(): array {
 
   public static function hook(Markup $markup, array &$output): string
   {
+    $config = \Drupal::config('coyote_img_desc.settings');
     $hostUri = $output['_coyote_node_url'];
+    $disableParsing = $config->get('disable_coyote_filtering');
+    if ($disableParsing) return $markup;
+   
+    $isPublished = $output['#node']->isPublished();
+    $coyote_process_unpublished_nodes = $config->get('coyote_process_unpublished_nodes');
```
Comment (jkva) [inline src/Hook/NodeViewAlterPostRenderHook.php:28]: Can this variable name be renamed as camelCase?

File: src/Form/CoyoteImgDescForm.php [hunk 2]
```diff
@@ -113,7 +146,7 @@ private function verifyResourceGroup(
         Util::getResourceGroupUri()
       );
       if (is_null($group)) {
-          $form['api_organization']['#field_suffix'] = $this->t("Resource group '@resourceGroup' could not be created", ['@resourceGroup' => Constants::RESOURCE_GROUP_NAME]);
+          $form['api_organization']['#field_suffix'] = $this->t("Resource group '@resourceGroup' could not be created", array('@resourceGroup' => Constants::RESOURCE_GROUP_NAME));
```
Comment (jkva) [inline src/Form/CoyoteImgDescForm.php:149]: Is the `array()` function over literal notation required for something?

File: src/Form/CoyoteImgDescForm.php [hunk 3]
```diff
@@ -82,9 +85,39 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
       $form['api_organization'] = self::getApiOrganizationIdFieldConfig($profile, $storedOrganizationId);
     }
 
+    $form['coyote_warning'] = [
+       '#type' => 'item',
+       '#markup' => '<div class="messages-list messages messages--warning">'.$this->t('Changing the API token or the endpoint or the Organization will clear the local plugin data')."</div>",
+       '#states' => array(
```
Comment (jkva) [inline src/Form/CoyoteImgDescForm.php:91]: Why the use of `array()` over literal notation?

**Review by jkva — CHANGES_REQUESTED** [review]
Some stylistic comments, this otherwise looks good. Feel free to remove the (pre-existing) use of `devel.dumper` if you have no need for it, I think it's something left over from my early development.

File: src/Hook/NodeViewAlterPostRenderHook.php (hunk 1 above)
Comment (catorghans) [reply src/Hook/NodeViewAlterPostRenderHook.php:28]: > Can this variable name be renamed as camelCase?

yup, will change

File: src/Form/CoyoteImgDescForm.php (hunk 2 above)
Comment (catorghans) [reply src/Form/CoyoteImgDescForm.php:149]: nope, personally I think array() is better readable, but opinions differ:
https://befused.com/php/short-array/

shall we use short array syntax everywhere?

Comment (jkva) [reply src/Form/CoyoteImgDescForm.php]: Yes please, I generally use this syntax across code, so if only for consistency's sake 👍

**Comment by jkva** [comment]
This closes #3, closes #5, closes #6, and closes #7.

**Pull request merged**